genie-web/src/main/java/com/netflix/genie/web/health/GenieCpuHealthIndicator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ *
+ *  Copyright 2016 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.health;
+
+import com.sun.management.OperatingSystemMXBean;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+
+import java.lang.management.ManagementFactory;
+
+/**
+ * Health indicator for system cpu usage. Mark the service out-of-service if it crosses the threshold.
+ * @author amajumdar
+ * @since 3.0.0
+ */
+@Component
+@Slf4j
+public class GenieCpuHealthIndicator implements HealthIndicator {
+    private static final String CPU_LOAD = "cpuLoad";
+    private final double maxCpuLoadPercent;
+    private final OperatingSystemMXBean operatingSystemMXBean;
+    /**
+     * Constructor.
+     *
+     * @param maxCpuLoadPercent The maximum physical memory threshold
+     */
+    @Autowired
+    public GenieCpuHealthIndicator(
+        @Value("${genie.health.cpu.maxLoadPercent:80}")
```

Comment:
I'm becoming increasingly uncomfortable using these `@Value` Strings. They're hard to hunt down when we want to change something and when we give them a default value like this it doesn't really break anything if we say change the name elsewhere this will still work. It might be better to create a `@ConfigurationProperties` class that at least binds at compile time. Also I would add whatever default value you have to the properties file so we have a record of it when we go to document.